atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java

```diff
@@ -197,6 +201,10 @@ private static SerializableTransactionManager create(
                 SimpleTimeDuration.of(config.getDefaultLockTimeoutSeconds(), TimeUnit.SECONDS));
         LockAndTimestampServices lockAndTimestampServices = createLockAndTimestampServices(
                 config,
+                JavaSuppliers.compose(runtimeConfig ->
+                                runtimeConfig.map(AtlasDbRuntimeConfig::timestampClient)
+                                        .orElse(ImmutableTimestampClientConfig.builder().build()),
```

Comment:
Ooh, yeah it will. Apparently Immutables supports this out of the box: https://immutables.github.io/immutable.html#singleton-instances - changed to use that.